### ISSUE TITLE
Messenger crashes when hangup is pressed

### ISSUE THREAD

**Issue by atsakiridis** [Mobicents/restcomm-android-sdk#162]
This, again is from Leon's testing. Scenario:
- Call 1235 using domain name
- Call takes a long time to be setup (and crash occurs before it does)

Analysis:

I would expect the crash to be because of NetworkOnMainThread, instead of an NPE. Seems like for some very strange reason we are trying to initialize signalling. This happens in only 2 cases:
- When reachability changes occur
- When Main Activity is created
So either Wifi was lost for a very short time leading to the shutdown and then reinitialization of facilities OR system was under pressure and decided to dispose Main activity (while on Call Activity). Then when hangup was pressed Main Activity was needed again and got recreated.

java.lang.NullPointerException
	at org.mobicents.restcomm.android.sipua.impl.sipmessages.Register.MakeRequest(Register.java:45)
	at org.mobicents.restcomm.android.sipua.impl.SipManager.Register(SipManager.java:328)
	at org.mobicents.restcomm.android.sipua.impl.DeviceImpl.Register(DeviceImpl.java:229)
	at org.mobicents.restcomm.android.client.sdk.RCDevice.initializeSignalling(RCDevice.java:222)
	at org.mobicents.restcomm.android.client.sdk.RCDevice.access$200(RCDevice.java:66)
	at org.mobicents.restcomm.android.client.sdk.RCDevice$2.run(RCDevice.java:251)
	at android.os.Handler.handleCallback(Handler.java:733)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:136)
	at android.app.ActivityThread.main(ActivityThread.java:5271)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:515)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:851)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:667)
	at dalvik.system.NativeStart.main(Native Method)

**Comment by atsakiridis** [comment]
I finally managed to reproduce it. This happens sporadically when WiFi becomes available and SDK is triggered to re-initialize networking facilities. It seems like the REGISTER is called before stack initialization has been completed

**Comment by atsakiridis** [comment]
Finally cracked this! Turns out that the JAIN listening point's underlying socket couldn't be bound because it was already in use. Discovered that it was due to the fact that JAIN initialization took place twice instead of once, because 2 reachability 'online' events where received. And even though I was keeping state to avoid this, it still happened because the RCDevice state was somehow corrupted withing the RCDevice broadcast receiver. I tried various ways around this, but the only one that worked was to make the state variable static.
